# Patch to add support for the pkgconfig-file sqlite3.pc, needed by at least firefox.
# See my upstream bugreport here http://www.sqlite.org/cvstrac/tktview?tn=3583.
# We want the amalgamation version of the sources, because we otherwise have to install
# tcl to build sqlite3 with special options like -DSQLITE_ENABLE_FTS3=1
#
# 2009-01-14 jue

diff -Nru sqlite-amalgamation-3.6.9.orig/Makefile.am sqlite-amalgamation-3.6.9/Makefile.am
--- sqlite-amalgamation-3.6.9.orig/Makefile.am	2009-01-14 14:13:11.000000000 +0100
+++ sqlite-amalgamation-3.6.9/Makefile.am	2009-01-14 17:04:30.000000000 +0100
@@ -12,3 +12,5 @@
 
 include_HEADERS = sqlite3.h sqlite3ext.h
 
+pkgconfigdir = ${libdir}/pkgconfig
+pkgconfig_DATA = sqlite3.pc
diff -Nru sqlite-amalgamation-3.6.9.orig/configure.ac sqlite-amalgamation-3.6.9/configure.ac
--- sqlite-amalgamation-3.6.9.orig/configure.ac	2009-01-14 14:13:11.000000000 +0100
+++ sqlite-amalgamation-3.6.9/configure.ac	2009-01-14 14:58:02.000000000 +0100
@@ -24,7 +24,8 @@
 # Check for library functions that SQLite can optionally use.
 AC_CHECK_FUNCS([fdatasync usleep fullfsync localtime_r gmtime_r])
 
-AC_CONFIG_FILES([Makefile])
+AC_CONFIG_FILES([Makefile sqlite3.pc])
+AC_SUBST(BUILD_CFLAGS)
 
 #-----------------------------------------------------------------------
 #   --enable-readline
diff -Nru sqlite-amalgamation-3.6.9.orig/sqlite3.pc.in sqlite-amalgamation-3.6.9/sqlite3.pc.in
--- sqlite-amalgamation-3.6.9.orig/sqlite3.pc.in	1970-01-01 01:00:00.000000000 +0100
+++ sqlite-amalgamation-3.6.9/sqlite3.pc.in	2009-01-14 12:49:13.000000000 +0100
@@ -0,0 +1,13 @@
+# Package Information for pkg-config
+
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: SQLite
+Description: SQL database engine
+Version: @VERSION@
+Libs: -L${libdir} -lsqlite3
+Libs.private: @LIBS@
+Cflags: -I${includedir}
